src/c_api.rs

```diff
@@ -177,6 +191,19 @@ pub fn json_api_get_double<M: Manager>(_: M, json: *const c_void, val: *mut c_do
             unsafe { *val = json.get_long() as f64 };
             Status::Ok as c_int
         }
+
+        SelectValueType::ULong => {
```

Comment:
Why would someone call get_double if the value is ulong? I see we did it for long as well but I wonder if its not an API missuse?

Reply:
Basically this is an optimization, for example, an array of doubles, in which some may be integers, not requiring to consult the type of each, or avoid [getDouble](https://github.com/RediSearch/RediSearch/blob/4387b045b764c7bfc802646dd0b8af52845b56b3/src/json.c#L143) to fail and then re-try as int.

Reply:
It could be done on the user side of the API, I believe API's should be minimal and simple.

Reply:
For the common use case, checking the type, and asking for the corresponding value is minimal and simple.
For double, it can also do more in case it is requested to (and `ULong` is now aligned with `Long`)
Do you think we should revert this optimization? (could affect RediSearch VecSim)